src/components/Builder/Builder.tsx

```diff
@@ -40,7 +49,7 @@ const Builder: FC = () => {
       {pathway && (
         <div className={styles.main}>
           <MuiThemeProvider theme={theme}>
-            <Sidebar />
+            <Sidebar currentNode={currentNode} />
```

Comment:
Is there a reason `Sidebar` and `DagreGraph` can't compute the `currentNode` themselves? They can get the pathway via the context and the nodeId via params

Reply:
the passed down version of currentNode is not the same as the version of currentNode retrieved from the pathway inside that component.